### PR TITLE
Add retry for JDBC tests

### LINKED ISSUE #815: JDBC tests needs to be retried
```
java.sql.SQLException: Region is unavailable

  at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:964)

  at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)

  at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)

  at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)

  at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)

  at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2483)

  at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2441)

  at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1381)

  at com.pingcap.tispark.datasource.BaseDataSourceTest.queryJDBC(BaseDataSourceTest.scala:213)

  at com.pingcap.tispark.datasource.BaseDataSourceTest.compareTiDBSelectWithJDBC(BaseDataSourceTest.scala:154)
```

### PR DESCRIPTION
**Pull request by birdstorm**
Close #815

### CODE REVIEW

File: core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala [hunk 1]
```diff
@@ -37,12 +34,6 @@ class BaseDataSourceTest(val table: String,
     initializeTimeZone()
   }
 
-  protected def initializeTimeZone(): Unit = {
-    tidbStmt = tidbConn.createStatement()
-    // Set default time zone to GMT-7
-    tidbStmt.execute(s"SET time_zone = '$timeZoneOffset'")
```
Comment (zhexuany) [inline core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala]: why remove this? This seems to be an irrelevant change.

Comment (birdstorm) [reply core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala]: useless since it is already available in BaseTiSparkSuite

File: core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala [hunk 2]
```diff
@@ -44,7 +44,7 @@ class BaseTiSparkTest extends QueryTest with SharedSQLContext {
   }
 
   protected def queryTiDB(query: String): List[List[Any]] = {
```
Comment (zhexuany) [inline core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala]: how about `queryTiDBUsingJDBC`?

Comment (birdstorm) [reply core/src/test/scala/org/apache/spark/sql/BaseTiSparkTest.scala]: Sounds reasonable.

File: core/src/test/scala/org/apache/spark/sql/QueryTest.scala [hunk 3]
```diff
@@ -401,6 +401,25 @@ abstract class QueryTest extends PlanTest {
       s"The physical plan has missing inputs:\n${query.queryExecution.executedPlan}"
     )
   }
+
+  protected def callWithRetry(execute: => ResultSet): ResultSet = {
```
Comment (zhexuany) [inline core/src/test/scala/org/apache/spark/sql/QueryTest.scala]: Some comments are nice to have.

Comment (birdstorm) [reply core/src/test/scala/org/apache/spark/sql/QueryTest.scala]: Sure

File: core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala (hunk 1 above)
Comment (zhexuany) [reply core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala]: Gotcha.

**Comment by zhexuany** [comment]
```
  org.apache.spark.sql.AnalysisException: Table or view 'testSelect_3554854253752661747_1560251825188' not found in database 'tidb_tispark_test';
``` 
is failed in `BasicSQLSuite` 

Please fix them.
